### PR TITLE
adding otel plugin

### PR DESCRIPTION
**Pull request by asteurer**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title`, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep $'\r' | wc -l` and expect 0 as a result)
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/api/hub/plugin_spin_otel.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Spin applications have the ability to export metrics and trace data. This plugin provides dashboards for viewing the data.
+
+## Requirements
+
+This plugin relies on third-party software to work properly. Please be sure you have the following installed before continuing:
+
+- Latest version of [Docker](https://www.docker.com/products/docker-desktop)
+
+# Installation
```
Comment (VamshiReddy02) [inline content/api/hub/plugin_spin_otel.md:28]:
```suggestion
## Installation
```

File: content/api/hub/plugin_spin_otel.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+## Install the canary version of the plugin
+
+The canary release of the command trigger plugin represents the most recent commits on `main` and may not be stable, with some features still in progress.
+
+```sh
+spin plugins install --url https://github.com/fermyon/otel-plugin/releases/download/canary/otel.json
+```
+
+# Usage
```
Comment (VamshiReddy02) [inline content/api/hub/plugin_spin_otel.md:49]:
```suggestion
## Usage
```

**Review by VamshiReddy02 — CHANGES_REQUESTED** [review]
Just a couple of tiny changes, then LGTM.

**Review by VamshiReddy02 — APPROVED** [review]

File: content/api/hub/plugin_spin_otel.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+title = "Spin OTel"
+template = "render_hub_content_body"
+date = "2024-08-16T00:22:56Z"
+content-type = "text/plain"
+tags = ["otel", "spin", "traces", "metrics", "logs", "observability"]
```
Comment (karthik2804) [inline content/api/hub/plugin_spin_otel.md:5]: I would limit it to 3 tags. The rest can be added as search terms using the `keywords` filed under the `[extra]` section.

ref: https://github.com/fermyon/developer/blob/a34891d88acb7c1bc519fdfdd3267d7374838211/content/api/hub/plugin_js2wasm.md?plain=1#L17

**Review by calebschoepp — CHANGES_REQUESTED** [review]
Please also update this [section](https://developer.fermyon.com/spin/v2/observing-apps#running-an-observability-stack) of the o11y docs to recommend your plugin as the best way to run an observability stack i.e. remove the instructions to directly run it in Docker.

File: content/api/hub/plugin_spin_otel.md (hunk 3 above)
Comment (itowlson) [reply content/api/hub/plugin_spin_otel.md:5]: Also `spin` is not a very useful tag in this context, because, well _gestures_

**Review by calebschoepp — APPROVED** [review]

**Pull request merged**